### PR TITLE
Feature/withDefaults

### PR DESCRIPTION
**Pull request by MrHen**
@justinmchase 
@adizam 

Adds a way to set default values at the step level.

### CODE REVIEW

File: lib/autoBlock.js [hunk 1]
```diff
@@ -29,48 +29,53 @@ var validBlockKeys = [
...
 
-  // Set config.done if not provided
-  if (!config.done) {
-    config.done = done || (() => {
-    })
+  var noop = () => {
   }
 
+  // Set config.done if not provided
+  config.done = config.done || done || noop
```
Comment (justinmchase) [inline lib/autoBlock.js:47]: Seems a little weird, what if both config.done and done are provided? It seems like done won't be called in that case.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "auto-block",
-  "version": "1.3.0",
+  "version": "1.4.0",
```
Comment (justinmchase) [inline package.json:3]: Non breaking additive changes?

**Review by justinmchase — APPROVED** [review]

File: lib/autoBlock.js (hunk 1 above)
Comment (MrHen) [reply lib/autoBlock.js:47]: Yeah, it's weird but it's [documented](https://github.com/TrackIF/auto-block#done).

File: package.json (hunk 2 above)
Comment (MrHen) [reply package.json:3]: Correct.

**Pull request merged**